ServerCore/Pages/Events/FastestSolves.cshtml.cs

```diff
@@ -52,7 +52,7 @@ public async Task OnGetAsync(TeamPuzzleSortOrder? teamPuzzleSort, SinglePlayerPu
         private async Task updateTeamPuzzleStats()
         {
             bool isRegisteredUser = await this.IsRegisteredUser();
-            if (!isRegisteredUser)
+            if (EventRole == EventRole.play && !isRegisteredUser)
```

Comment:
For authors, this would need to filter the puzzles to the author's puzzles

Reply:
Shouldn't authors see the fastest solves even if they didn't write the puzzle?
After all, players can see the fastest solves of all puzzles

Reply:
As a player, the list of puzzles shown is filtered to what you have unlocked, so I agree with Morgan.

Reply:
Is there a reason we don't want authors to see the fastest solves page of all the puzzles?

Reply:
Exactly

Reply:
Makes sense. I updated it